### PR TITLE
chore(localdebug): add telemetry to transparent task

### PR DESCRIPTION
**Pull request by XiaofuHuang**
1. Add new event to each task. If user changed the input args, the value will be `<unknown>`.
2. If user do not remove `debug-check-prerequisite` task, the `correlation-id` is not empty and `debug-all` event contains the error info and duration.
3. If user remove  `debug-check-prerequisite` task, the `debug-prelaunch-task-info` can be used to calculate the debug trace.

TODO: add telemetry to SPFx project

### CODE REVIEW

File: packages/vscode-extension/src/debug/localTelemetryReporter.ts [hunk 1]
```diff
@@ -117,3 +138,113 @@ export async function sendDebugAllEvent(
...
+        });
+      }
+      return dependsOnArr;
+    };
+    return {
+      m365Overall: getDependsOn(TaskLabel.M365Overall),
+      overall: getDependsOn(TaskLabel.Overall),
+    };
+  } catch (error) {
+    console.log(error);
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/localTelemetryReporter.ts:241]: Why console.log? Remove this?

File: packages/vscode-extension/src/debug/localTelemetryReporter.ts [hunk 2]
```diff
@@ -117,3 +138,113 @@ export async function sendDebugAllEvent(
...
+  type: string;
+  command: string;
+}
+
+interface IPreLaunchTaskInfo {
+  m365Overall?: IDependsOn[];
+  overall?: IDependsOn[];
+}
+
+export async function getPreLaunchTaskInfo(): Promise<IPreLaunchTaskInfo | undefined> {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/localTelemetryReporter.ts:199]: Add tests to cover:
- happy path
- error reading tasks.json
- user data is masked correctly

File: packages/vscode-extension/src/debug/localTelemetryReporter.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/localTelemetryReporter.ts]: Sorry, removed it.

File: packages/vscode-extension/src/debug/localTelemetryReporter.ts (hunk 2 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/localTelemetryReporter.ts:195]: Added ut for `getPreLaunchTaskInfo`

**Review by swatDong — DISMISSED** [review]

**Comment by a1exwang** [comment]
Is this needed to be added here? `checkAndInstall()` is only for old local debug?
```
[TelemetryProperty.DebugIsTransparentTask]: "false",
```

https://github.com/OfficeDev/TeamsFx/blob/51d8ffe8468c851b3d0dad32bc9684e192588b73/packages/vscode-extension/src/debug/prerequisitesHandler.ts#L332

**Review by HuihuiWu-Microsoft — DISMISSED** [review]

**Review by a1exwang — DISMISSED** [review]

**Comment by XiaofuHuang** [comment]
> Is `[TelemetryProperty.DebugIsTransparentTask]: "false"` needed to be added here? `checkAndInstall()` is only for old local debug, so `DebugIsTransparentTask` needs to be explicitly set, otherwise it will default to `true`?
> 
> https://github.com/OfficeDev/TeamsFx/blob/51d8ffe8468c851b3d0dad32bc9684e192588b73/packages/vscode-extension/src/debug/prerequisitesHandler.ts#L332
Added [TelemetryProperty.DebugIsTransparentTask]: "false".

**Review by a1exwang — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Pull request merged**